cluster/addons/dns/kube2sky/kube2sky.go

```diff
@@ -105,21 +134,30 @@ type kube2sky struct {
 // Removes 'subdomain' from etcd.
 func (ks *kube2sky) removeDNS(subdomain string) error {
 	glog.V(2).Infof("Removing %s from DNS", subdomain)
-	resp, err := ks.etcdClient.RawGet(skymsg.Path(subdomain), false, true)
+	ctx := newContext()
+	opts := &etcd.GetOptions{
+		Recursive: true,
+	}
+	resp, err := ks.etcdKeysAPI.Get(ctx, skymsg.Path(subdomain), opts)
 	if err != nil {
 		return err
 	}
-	if resp.StatusCode == http.StatusNotFound {
+	if resp.Node == nil {
```

Comment:
@timothysc you did the go-etcd conversion for apiserver... this look right to you?